### PR TITLE
Fix handling of transient contracts in extractor

### PR DESCRIPTION
**Pull request by cocreature**
changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: extractor/src/main/scala/com/digitalasset/extractor/ledger/types/TransactionTree.scala [hunk 1]
```diff
@@ -12,14 +12,24 @@ import java.time.Instant
...
+// root node (e.g. because the root nodes were all for different templates)
+// and the child event ids can point to nodes that have been filtered out.
 final case class TransactionTree(
     transactionId: String,
     workflowId: String,
     effectiveAt: Instant,
     offset: String,
-    events: Map[String, Event],
-    rootEventIds: Set[String]
+    // Ordered in pre-order (the order the events happen on the ledger).
```
Comment (aherrmann-da) [inline extractor/src/main/scala/com/digitalasset/extractor/ledger/types/TransactionTree.scala:30]:
```suggestion
    // Ordered in pre-order (the order in which the events happen on the ledger).
```

File: extractor/src/test/suite/scala/com/digitalasset/extractor/TransactionMultiTableSpec.scala [hunk 2]
```diff
@@ -63,45 +63,62 @@ class TransactionMultiTableSpec
...
+    exercise1.child_event_ids.asArray.toList.toVector.flatten should contain(event_id_accept.asJson)
+    transaction_id_accept shouldEqual transaction2.transaction_id
     // which is not archived
-    archived_by_transaction_id2 shouldEqual None
-    archived_by_event_id2 shouldEqual None
+    archived_by_transaction_id_accept shouldEqual None
+    archived_by_event_id_accept shouldEqual None
+
+    // `transaction3` created `contract3`; then
+    transaction_id2 shouldEqual transaction3.transaction_id
```
Comment (aherrmann-da) [inline extractor/src/test/suite/scala/com/digitalasset/extractor/TransactionMultiTableSpec.scala:114]: How does the comment relate to the check?

Comment (cocreature) [reply extractor/src/test/suite/scala/com/digitalasset/extractor/TransactionMultiTableSpec.scala]: The naming here is horrible:
The multitable schema creates one table per schema. `transaction_id2` is transaction id of the second offer contract which is the third contract in total. I’ve added the `offer` suffix consistently and a few comments which hopefully help.

Comment (aherrmann-da) [reply extractor/src/test/suite/scala/com/digitalasset/extractor/TransactionMultiTableSpec.scala]: Thank you for clarifying, makes sense.

**Review by aherrmann-da — APPROVED** [review]
LGTM

**Pull request merged**